Running 'nuxt build' when '@prisma/nuxt' is listed in the modules in nuxt.config deletes other nuxt modules

## Versions
```json
"@prisma/client": "^5.17.0"
"@prisma/nuxt": "^0.0.30"
"nuxt": "^3.12.3",
"prisma": "^5.17.0",
```

## Issue

Running the dev server works fine.
Running `nuxt build` (via `npm run build`) when `@prisma/nuxt` is listed in the modules in `nuxt.config.ts` deletes other nuxt modules from `node_modules/@nuxt`, leading to errors such as:

```
Error while requiring module @nuxt/devtools: Error: Cannot find module '@nuxt/devtools'
```

or '@nuxt/test-utils', etc: the first module that fails to be found depends on the order in which modules are listed in nuxt.config.ts.

Before running `npm run build`, the nuxt part of my node_modules looked like this:

![Screenshot from 2024-07-18 14-54-08](https://github.com/user-attachments/assets/7d5e11de-eb9d-40ab-8029-85de4162795a)

After running `npm run build`, which errored as above, the nuxt part of my node_modules looked like this:

![Screenshot from 2024-07-18 14-53-30](https://github.com/user-attachments/assets/0a58aa8a-731d-492e-8637-747e79df2aa8)

This uninstalled or deleted a lot of nuxt modules, such that for example trying to run `npm run build` immediately afterwards led to:

```bash
npm run build

> build
> nuxt build

sh: 1: nuxt: not found
```

The other modules listed in my `nuxt.config.ts` are:

```ts
modules: [
    "@nuxtjs/fontaine",
    "@pinia/nuxt",
    "@nuxt/test-utils/module",
    "@nuxt/eslint",
    "@nuxtjs/i18n",
  ],
```

## Possible workaround?
My workaround is to not use `@prisma/nuxt`.

I found that I could run the build step and it did not delete nuxt modules, and I could still use prisma (at least, I could perform operations from within an api endpoint, as below) if I uninstalled `@prisma/nuxt`, omitted the module from nuxt.config.ts, and used '[usage option B](https://www.prisma.io/docs/orm/more/help-and-troubleshooting/help-articles/prisma-nuxt-module#option-b-libprismats)' instead of the composable.

```ts
import prisma from "~/lib/prisma";

export default defineEventHandler(async (event) => {
  const uniqueString = Math.random().toString(36).substring(7);

  await prisma.user.create({
    data: {
      email: `${uniqueString}@email.com`,
      name: "Hello",
    },
  });

  // Return all users
  return {
    user: await prisma.user.findMany(),
  };
});
```

```
# package.json
...
  "devDependencies": {
    "@antfu/eslint-config": "2.21.0",
    "@nuxt/eslint": "^0.3.13",
    "@nuxt/test-utils": "^3.13.1",
    "@nuxtjs/i18n": "^8.3.1",
    "@playwright/test": "^1.45.1",
    "@prisma/client": "^5.17.0",
    "@types/lodash.throttle": "^4.1.9",
    "@types/node": "^20.14.9",
    "@vitest/coverage-v8": "^1.6.0",
    "@vue/test-utils": "^2.4.6",
    "jsdom": "^24.1.0",
    "lint-staged": "^15.2.7",
    "nuxt": "^3.12.3",
    "prisma": "^5.17.0",
    "sass": "^1.77.6",
    "simple-git-hooks": "^2.11.1",
    "vitest": "^1.6.0"
  },
```

--- a/.github/workflows/preview-publish.yml
+++ b/.github/workflows/preview-publish.yml
@@ -1,4 +1,4 @@
-name: Preview Release on PR
+name: Publish Experimental Release
 on:
   pull_request:
     branches:
@@ -6,8 +6,8 @@ on:
 permissions:
   contents: write # Required for publishing releases
 jobs:
-  preview-release:
-    name: Preview Dev Release
+  experimental-release:
+    name: Experimental Dev Release
     runs-on: ubuntu-latest
     permissions:
       contents: write # For publishing to GitHub releases
@@ -56,20 +56,12 @@ jobs:
       - name: Audit Dependencies
         run: npm audit signatures
 
-      # Step 10: Generate and log preview release version
-      - name: Generate and Log Preview Release Version
-        id: preview_version
-        run: |
-          VERSION="0.0.0-pr-${{ github.event.pull_request.number }}-$(git rev-parse --short HEAD)"
-          echo "Preview release version: $VERSION"
-          echo "VERSION=$VERSION" >> $GITHUB_ENV
+      # Step 10: Set experimental version
+      - name: Set Experimental Version
+        run: npm version --no-git-tag-version 0.0.0-experimental-${GITHUB_SHA::7}
 
-      # Step 11: Create a preview release
-      - name: Publish Preview Release
+      # Step 11: Publish Experimental Release
+      - name: Publish Experimental Release
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          NPM_TOKEN: ${{ secrets.NPM_NUXT_TOKEN }}
-          VERSION: ${{ env.VERSION }}
-        run: |
-          echo "Publishing preview release $VERSION"
-          npx semantic-release --no-ci --branches main --tag-format "v${VERSION}"
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_NUXT_TOKEN }}
+        run: npm publish --access public --tag experimental

--- a/.github/workflows/preview-publish.yml
+++ b/.github/workflows/preview-publish.yml
@@ -62,13 +62,14 @@ jobs:
         run: |
           VERSION="0.0.0-pr-${{ github.event.pull_request.number }}-$(git rev-parse --short HEAD)"
           echo "Preview release version: $VERSION"
-          echo "version=$VERSION" >> $GITHUB_ENV
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
 
       # Step 11: Create a preview release
       - name: Publish Preview Release
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           NPM_TOKEN: ${{ secrets.NPM_NUXT_TOKEN }}
+          VERSION: ${{ env.VERSION }}
         run: |
-          echo "Publishing preview release ${{ env.version }}"
-          npx semantic-release --no-ci --branches main --tag-format "v${{version}}-pr-${{ github.event.pull_request.number }}-$(git rev-parse --short HEAD)"
+          echo "Publishing preview release $VERSION"
+          npx semantic-release --no-ci --branches main --tag-format "v${VERSION}"

--- a/.github/workflows/preview-publish.yml
+++ b/.github/workflows/preview-publish.yml
@@ -71,5 +71,4 @@ jobs:
           NPM_TOKEN: ${{ secrets.NPM_NUXT_TOKEN }}
         run: |
           echo "Publishing preview release ${{ env.version }}"
-          # Use semantic-release or another tool to publish the preview version
-          npx semantic-release --no-ci --branches main --tag-format "v${{ env.version }}"
+          npx semantic-release --no-ci --branches main --tag-format "v${{version}}-pr-${{ github.event.pull_request.number }}-$(git rev-parse --short HEAD)"

--- a/.github/workflows/preview-publish.yml
+++ b/.github/workflows/preview-publish.yml
@@ -1,0 +1,75 @@
+name: Preview Release on PR
+on:
+  pull_request:
+    branches:
+      - main
+permissions:
+  contents: write # Required for publishing releases
+jobs:
+  preview-release:
+    name: Preview Dev Release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # For publishing to GitHub releases
+      id-token: write # For npm provenance
+    steps:
+      # Step 1: Checkout the repository
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      # Step 2: Setup Node.js environment
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "lts/*"
+          cache: "npm"
+
+      # Step 3: Configure npm for authentication
+      - name: Configure npm for Authentication
+        run: |
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_NUXT_TOKEN }}" >> ~/.npmrc
+          echo "@prisma:registry=https://registry.npmjs.org/" >> ~/.npmrc
+
+      # Step 4: Install dependencies
+      - name: Install Dependencies
+        run: npm clean-install
+
+      # Step 5: Prepare project
+      - name: Prepare Project
+        run: npx nuxi prepare
+
+      # Step 6: Build project
+      - name: Build Project
+        run: npm run dev:build
+
+      # Step 7: Lint the code
+      - name: Lint Code
+        run: npm run lint
+
+      # Step 8: Prepack the project
+      - name: Prepack
+        run: npm run prepack
+
+      # Step 9: Audit dependencies
+      - name: Audit Dependencies
+        run: npm audit signatures
+
+      # Step 10: Generate and log preview release version
+      - name: Generate and Log Preview Release Version
+        id: preview_version
+        run: |
+          VERSION="0.0.0-pr-${{ github.event.pull_request.number }}-$(git rev-parse --short HEAD)"
+          echo "Preview release version: $VERSION"
+          echo "version=$VERSION" >> $GITHUB_ENV
+
+      # Step 11: Create a preview release
+      - name: Publish Preview Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_NUXT_TOKEN }}
+        run: |
+          echo "Publishing preview release ${{ env.version }}"
+          # Use semantic-release or another tool to publish the preview version
+          npx semantic-release --no-ci --branches main --tag-format "v${{ env.version }}"

--- a/src/module.ts
+++ b/src/module.ts
@@ -214,14 +214,7 @@ export default defineNuxtModule<PrismaExtendedModule>({
         return;
       }
 
-      const promptResults = await executeRequiredPrompts({
-        promptForMigrate: false && !skip_all_prompts,
-        promptForPrismaStudio: false && !skip_all_prompts,
-      });
-
-      if (promptResults?.promptForInstallingStudio) {
-        await installAndStartPrismaStudio();
-      }
+      await installAndStartPrismaStudio();
     };
 
     if (!prismaSchemaExists) {

--- a/src/module.ts
+++ b/src/module.ts
@@ -88,10 +88,12 @@ export default defineNuxtModule<PrismaExtendedModule>({
       addServerImportsDir(resolver(runtimeDir, "utils"));
       addServerImportsDir(resolver(runtimeDir, "server/utils"));
 
-      nuxt.options.vite.optimizeDeps ||= {};
-      nuxt.options.vite.optimizeDeps = {
-        include: ["@prisma/nuxt > @prisma/client"],
-      };
+      nuxt.options.vite.optimizeDeps = defu(
+        nuxt.options.vite.optimizeDeps || {},
+        {
+          include: ["@prisma/nuxt > @prisma/client"],
+        },
+      );
     };
 
     const force_skip_prisma_setup =

--- a/src/module.ts
+++ b/src/module.ts
@@ -216,7 +216,7 @@ export default defineNuxtModule<PrismaExtendedModule>({
 
       const promptResults = await executeRequiredPrompts({
         promptForMigrate: false && !skip_all_prompts,
-        promptForPrismaStudio: true && !skip_all_prompts,
+        promptForPrismaStudio: false && !skip_all_prompts,
       });
 
       if (promptResults?.promptForInstallingStudio) {

